### PR TITLE
nav meshes managing

### PR DESCRIPTION
**Pull request by RogerNogue**
now navigation meshes are stored in Resources/NavigationMeshes.
To test:
- check that the graveyard navmesh is in Resources/NavigationMeshes
- start the engine loading the mentioned scene and check that the navMesh works

### CODE REVIEW

**Comment by AlbertVVila** [comment]
@RogerNogue Is it possible to have more than one nav mesh now with this PR? If so, i'll test it with the other tests you mentioned.

File: Source/ModuleNavigation.cpp [hunk 1]
```diff
@@ -82,15 +82,29 @@ void ModuleNavigation::SaveConfig(JSON * config)
...
+		cleanValuesPOST();
+		cleanValuesPRE();
+		return;
+	}
 
+	std::stringstream path;
+	path << "Resources/NavigationMeshes/navMesh" << sceneName << ".bin";
 	char* navData2 = 0;
-	App->fsystem->Load("Assets/navigationMesh.bin", &navData2);
+	App->fsystem->Load(path.str().c_str(), &navData2);
```
Comment (Rubetoman) [inline Source/ModuleNavigation.cpp]: Shouldn't this be path.c_str()only? Instead of path.str().c_str()?

**Comment by RogerNogue** [comment]
> @RogerNogue Is it possible to have more than one nav mesh now with this PR? If so, i'll test it with the other tests you mentioned.

yes, now each scene has its own nav mesh

File: Source/ModuleNavigation.cpp (hunk 1 above)
Comment (RogerNogue) [reply Source/ModuleNavigation.cpp]: no, because path is an std::stringstream, not an std::string

**Comment by RogerNogue** [comment]
I noticed some possible errors that could appear if the json of the scene is lost or modified or something, so I made sure that this does not happen storing some extra information in an additional file

**Comment by EnriquePuyol** [comment]
Could it be possible to create a scene with the player ready to be moved with the new nav mesh?

**Comment by RogerNogue** [comment]
> Could it be possible to create a scene with the player ready to be moved with the new nav mesh?

When a new scene is created, it needs a nav mesh attached to this new scene. Once a scene has a nav mesh attached, it stays there. It stays as 2 files in Resources/NavigationMeshes

**Pull request merged**